classes/Launchpad.sc

```diff
@@ -123,13 +123,13 @@ Launchpad{
 LPRange{
 	var <>pads;
 	*fromPadRange{|p, fx, fy, tx, ty|
-		var newPads = Array2D.new;
-		(fx..fy).do{|f, fi|
-			(tx..ty).do{|t, ti|
-				newPads.put(fi, ti, p.at(f, t));
+		var newPads = Array2D.new(1 + (ty - fy), 1 + (tx - fx));
+		(fx..tx).do{|x, xi|
+			(fy..ty).do{|y, yi|
+				newPads.put(yi, xi, p.at(x, y));
```

Comment:
I realized that I implemented the last line as: newPads.put(xi, yi, p.at(x, y)); ... and I also remember that I wondered why I choose it is way. Does your implementation work for you?

Reply:
ok how to deal with this ... my instruments are all implemented with swapped x y ... would this be very confusing? ... ok lets use your implementation, because it makes more sense. But maybe some things will behave quite strange and would need some update as well.

Reply:
True, I did swap the yi and xi here. It throws an out of range error if not. This way it works now with my launchpad with various ranges I've been trying out. However, I think it actually is transposing so rows become columns. It was probably throwing the range error due to the swapped values in the Array2D.new, which I probably should have swapped there instead of the put statement....

Reply:
And, swapped.